### PR TITLE
New: task edit and delete

### PR DESCRIPTION
**Pull request by DanDeMicco**

### CODE REVIEW

File: src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js [hunk 1]
```diff
@@ -117,28 +117,141 @@ class ActivityFeed extends React.Component<Props, State> {
...
+            id
+        );
+    };
+
+    /**
+     * Called on failed update of a task
+     *
+     * @param {Object} e the error
+     */
+    updateTaskErrorCallback = (e: Error, id: string) => {
```
Comment (DanDeMicco) [inline src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js]: can probably combine this and below function into a single function e.g. ```taskErrorCallback```

**Comment by boxcla** [comment]
# Webpack Bundle Size Github Bot

| Bundle        | Diff | Percentage Diff | Before | After |
| ------------- |:----:|:---------------:|:------:|:-----:|
| Total | 0 B | 0.00% | 0 B | 0 B |

Compared against previously merged PR #338

**Powered by [webpack-bundle-size-github-bot](https://github.com/wenboyu2/webpack-bundle-size-github-bot/)**

File: src/components/ContentSidebar/ContentSidebar.js [hunk 2]
```diff
@@ -645,6 +645,126 @@ class ContentSidebar extends PureComponent<Props, State> {
...
+     * Task update success callback
+     *
+     * @private
+     * @param {Object} task - Box task
+     * @return {void}
+     */
+    onTaskUpdateSuccessCallback(task: Task) {
+        const { tasks } = this.state;
+
+        if (tasks) {
```
Comment (priyajeet) [inline src/components/ContentSidebar/ContentSidebar.js]: See my comment on #332

File: src/components/ContentSidebar/ContentSidebar.js [hunk 3]
```diff
@@ -699,6 +699,190 @@ class ContentSidebar extends PureComponent<Props, State> {
...
+     * @private
+     * @param {Object} task - Box task
+     * @return {void}
+     */
+    deleteTaskSuccessCallback = (taskId: string) => {
+        const { tasks } = this.state;
+
+        if (tasks) {
+            const { entries } = tasks;
+
```
Comment (priyajeet) [inline src/components/ContentSidebar/ContentSidebar.js]: You love them blank lines, don't you. Feel a bit random, to have it between some const declarations and not others.

Comment (DanDeMicco) [reply src/components/ContentSidebar/ContentSidebar.js]: Yeah I guess typically I put blank lines in between variable declarations and if statements/function calls. Other pieces of the code have random spaces as well. Is there something that can be done in eslint / prettier to make this consistent?

File: src/components/ContentSidebar/ContentSidebar.js [hunk 4]
```diff
@@ -699,6 +699,190 @@ class ContentSidebar extends PureComponent<Props, State> {
...
+    };
+
+    /**
+     * Task update success callback
+     *
+     * @private
+     * @param {Object} task - Box task
+     * @return {void}
+     */
+    updateTaskSuccessCallback(task: Task) {
```
[the remark this answers is not in the document]
Comment (DanDeMicco) [reply src/components/ContentSidebar/ContentSidebar.js]: Good catch!

**Pull request merged**